contributors/design-proposals/optional-configmap.md

```diff
@@ -0,0 +1,111 @@
+# Optional ConfigMaps
+
+## Goal
+
+Allow the ConfigMaps that are used to populate the environment variables of a
+container and files within a Volume to be optional.
+
+## Use Cases
+
+An application developer can define a set of environment variables for a
+container by referencing a ConfigMap with default values. The developer can
+also define an optional ConfigMap that is used to override the values. There
+may be a set of these optional ConfigMaps that provide environment specific
+overrides for a given test cluster or region cluster. If no overrides are
+present, the container will get the default values.
+
+An application developer can populate a volume with files defined from a
+ConfigMap. The developer may have some required files to be created and have
+optional additional files at a different target. The developer can specify on
+the Pod that there is an optional ConfigMap that will provide these additional
+files if present.
+
+## Design Points
+
+A container can specify a set of existing ConfigMaps to populate environment
+variables. These ConfigMaps can be marked as optional. The default behavior
+will cause the container to fail to start when the ConfigMap is not defined.
+If the ConfigMap is marked optional, when the ConfigMap is not defined, the
+container will skip this ConfigMap and continue.
+
+A pod can specify a set of Volumes to mount. A ConfigMap can represent the
+files to populate the volume. The ConfigMaps should be allowed to be optional.
+The default behavior will fail the volume setup if the ConfigMap is missing.
+If the ConfigMap is marked as optional, the volume will be set up with no
+files, and continue.
```

Comment:
Currently, the files in the configmap volume are updated from the configmap while the pod is running. 

Define what happens if an optional configmap is missing when the pod is started on the node, and the configmap appears later while the pod is running... will the files be populated at that point, or not?